### PR TITLE
CASSANDRA-16527 Fix flaky ValidationExecutorTest

### PR DESCRIPTION
**Pull request by bereng**

### CODE REVIEW

**Comment by bereng** [comment]
CI [j11](https://app.circleci.com/pipelines/github/bereng/cassandra/233/workflows/0cdd8f63-dd07-4a7f-a8f8-21075ad51c08)
CI [j8](https://app.circleci.com/pipelines/github/bereng/cassandra/233/workflows/f745e881-e832-4c30-b138-fcf79dc6441d)

File: test/unit/org/apache/cassandra/db/compaction/ValidationExecutorTest.java [hunk 1]
```diff
@@ -79,7 +80,16 @@ public void testQueueOnValidationSubmission() throws InterruptedException
...
+        // getActiveTaskCount() relies on getActiveCount() which gives an approx number so we poll it
+        boolean activeTasksOk = false;
+        for (int i = 0; i < 100 && !activeTasksOk; i++)
+        {
+            TimeUnit.MILLISECONDS.sleep(10);
+            activeTasksOk = validationExecutor.getActiveTaskCount() == 2;
+        }
+        if (!activeTasksOk)
+            fail();
+
```
Comment (blerer) [inline test/unit/org/apache/cassandra/db/compaction/ValidationExecutorTest.java]: The project as already an utility to perform that kind of check `Util.spinAssertEquals`. To see an example you can have a look into `ThreadPoolMetricsTest `

Comment (bereng) [reply test/unit/org/apache/cassandra/db/compaction/ValidationExecutorTest.java]: +1 TIL. Passes 10K runs locally.

File: test/unit/org/apache/cassandra/db/compaction/ValidationExecutorTest.java [hunk 2]
```diff
@@ -30,11 +30,13 @@
 import org.junit.Before;
 import org.junit.Test;
 
+import org.apache.cassandra.Util;
 import org.apache.cassandra.config.DatabaseDescriptor;
 import org.apache.cassandra.utils.concurrent.SimpleCondition;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/db/compaction/ValidationExecutorTest.java]: Nit: unused import after using `spinAssertEquals`.

**Review by adelapena — APPROVED** [review]
Looks good to me. I have reproduced the problem with a few tens of thousands local runs and confirmed that the patch solves it, at least locally. Unfortunately I think it's not practical to use our internal test multiplexer because, if I'm right, we need a very high number of runs to reproduce the original problem, and the multiplexer seems to add an overhead of ~5s per test run.

File: test/unit/org/apache/cassandra/db/compaction/ValidationExecutorTest.java (hunk 2 above)
Comment (bereng) [reply test/unit/org/apache/cassandra/db/compaction/ValidationExecutorTest.java]: Pushed fixed imports thx!

**Pull request closed without merging**